pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2541,7 +2541,25 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
   public ResultSet getFunctions(String catalog, String schemaPattern, String functionNamePattern)
       throws SQLException {
-    return getProcedures(catalog, schemaPattern, functionNamePattern);
+    String sql;
+    sql = "SELECT NULL AS FUNCTION_CAT, n.nspname AS FUNCTION_SCHEM, p.proname AS FUNCTION_NAME, "
+            + "d.description AS REMARKS, "
+            + DatabaseMetaData.procedureReturnsResult + " AS FUNCTION_TYPE, "
```

Comment:
Sounds good to me. I checked the PostgreSQL doc and 'record' type seems to be appropriate. They also mentioned "Row Type", but I don't see that as possible result from the `pg_proc` table. However I do see `unknown` as possible format_type result too. Maybe this can be the `functionResultUnknown` case?